### DISCUSSION TITLE
oracledb.exceptions.NotSupportedError: DPY-3010: connections to this database server version are not supported by python-oracledb in thin mode

### DISCUSSION THREAD

**Discussion by ckalas** [oracle/python-oracledb#186] [General]
I'm struggling to get a native installation on arm macOS connecting to an Oracle Database 11g Enterprise Edition Release 11.2.0.4.0 - 64bit

Are there any ways to get patch the thin client which ships with the library or somehow make this work?

**Comment by anthony-tuininga** [comment]
Unfortunately, no. The 11g database requires considerably different data to be sent between the database and the client, so the patches would be considerable. Since 11g is no longer supported in any meaningful way we decided not to spend time adding support for such an old database! You will have to use thick mode or upgrade the database.

**Reply by ckalas** [reply]
okay, so realistically i will need to run the stuff through rosetta then or give up and dockerize my dev environment?

**Reply by anthony-tuininga** [reply]
Realistically, yes. :-)
